### PR TITLE
Feature: Subsite compatibility #3

### LINKED ISSUE #3: Feature: Subsite compatibility
Changes required to make the module compatible with subsites is relatively minor and should therefore be combined in the main module.
**Expected behaviour**
- If the subsites module is not installed there should be no impact on existing functions
- If the subsites module is installed all indexed classes that are linked to a subsite should be indexed in a way that identifies which subsite they are from
- There should be a config option that can be set in yaml to set whether all subsite results are displayed together, or if only the current subsite results appear
- Documentation should be updated to explain the use of the config option, including the default behaviour if it is not set

### PR DESCRIPTION
**Pull request by sig-peggy**
Meeting requirements identified in #3 plus additional functions:

- Behaviour for indexed classes which do not have a SubsiteID field - these appear on all subsites
- Behaviour when Files are indexed which generally do not have their own SubsiteID set but can inherit view permissions based on the parent Folder SubsiteID:
  - Optional File extension to get the SubsiteID from the parent Folder so that Files that can't be viewed on a subsite will not appear in the results count
  - Show main site Files across all subsites (to align with other Subsite module behaviour) unless a config option is changed to prevent this

### CODE REVIEW

File: src/Extensions/FileSubsiteSearchExtension.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * @author Signify Ltd <info@signify.co.nz>
+ * @copyright Copyright (c) 2024 - now() Signify Ltd
+ */
+
+namespace Firesphere\SolrSearch\Extensions;
+
+use SilverStripe\Assets\Folder;
+use SilverStripe\Core\Extension;
+
+class FileSubsiteSearchExtension extends Extension
```
Comment (sig-sam) [inline src/Extensions/FileSubsiteSearchExtension.php:25]: minor: Missing a docblock.

File: src/Extensions/SubsitesDocumentFactoryExtension.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace Firesphere\SolrSearch\Extensions;
+
+use Firesphere\SolrSearch\States\SubsiteState;
+use SilverStripe\Core\Extension;
+
+/**
+ * Update Documents per locale
+ */
+class SubsitesDocumentFactoryExtension extends Extension
```
Comment (sig-sam) [inline src/Extensions/SubsitesDocumentFactoryExtension.php:25]: minor: Add `@package` and `@property` annotations for consistency.

**Review by sig-sam — CHANGES_REQUESTED** [review]
There are a couple of minor documentation issues. Otherwise, it looks pretty good. I can see that all the expected behaviour has been implemented.

If you haven't already considered this, for QA testing, it might be good to deploy to two sites: one with subsites and the other without.

**Comment by sig-peggy** [comment]
Fixed noted doc blocks.
I have got a plan for testing with and without subsites generally, but the one lined up with subsites doesn't cover any other kinds of classes besides SiteTree so I have been pondering the best way to cover those parts in QA.

**Review by sig-sam — APPROVED** [review]

File: src/Extensions/FileSubsiteSearchExtension.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+/**
+ * Class FileSubsiteExtension|Firesphere\SolrSearch\Extensions\FileSubsiteExtension
+ *
+ * @package Firesphere\Solr
+ * @author Signify Ltd <info@signify.co.nz>
+ * @copyright Copyright (c) 2024 - now() Signify Ltd
```
Comment (sig-steve) [inline src/Extensions/FileSubsiteSearchExtension.php]: I suggest **not** using the cutesy `2024 - now()`. Lawyers are not known for their appreciation of cuteness.

I think this is sufficient:

```
@copyright Copyright 2024 Signify Ltd
```

**Comment by sig-steve** [comment]
There are other occurrences of the "Copyright (c) 2024 - now() Signify Ltd" that I'd like to see changed. I only noted the first.

**Comment by sig-peggy** [comment]
Updated the 2 Signify copyright lines - the other copyright lines added in this work are Firesphere ones in files that were copied from their subsites module as a starting point and then changed to meet our needs. @sig-steve I assume we still want to leave their copyright messages alone even if they are more 'cutesy' than we'd prefer?

**Comment by sig-steve** [comment]
Yes. We mustn't touch their copyright statements. Sorry, I mistakenly thought there was 3 Signify copyrights, not two. Thanks @sig-peggy!

**Comment by sig-peggy** [comment]
Testing completed on subsites with combined search results, subsites with separate search results, and a site without subsites. All tests passed.

**Pull request merged**